tests/integration/components/join-channel/component-test.js

```diff
@@ -0,0 +1,50 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, select } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import { A } from '@ember/array';
+import Service from '@ember/service';
+import { ircAccount, xmppAccount } from '../../../fixtures/accounts';
+
+class comsStub extends Service {
+  accounts = A();
+}
+
+module('Integration | Component | join-channel', function(hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function() {
+     this.owner.register('service:coms', comsStub);
+  });
+
+  test('Account menu for one account', async function(assert) {
+    this.coms = this.owner.lookup('service:coms');
+    this.coms.accounts.pushObject(ircAccount);
+    await render(hbs`<JoinChannel />`);
+
+    assert.equal(this.element.querySelectorAll('select#account option').length, 1, 'only has one option');
+    const el = this.element.querySelector('select#account option');
+    assert.equal(el.innerText, ircAccount.id, 'renders the only account option');
+    assert.ok(this.element.querySelector('select#account').disabled, 'is disabled');
+
+  });
+
+  test('Account menu for multiple accounts', async function(assert) {
+    this.coms = this.owner.lookup('service:coms');
+    this.set('coms.accounts', [ ircAccount, xmppAccount ]);
+    await render(hbs`<JoinChannel />`);
+
+    assert.equal(this.element.querySelectorAll('select#account option').length, 2, 'renders options for all accounts');
+    assert.notOk(this.element.querySelector('select#account').disabled, 'is not disabled');
+  });
+
+  test('Select different account', async function(assert) {
+    this.coms = this.owner.lookup('service:coms');
+    this.set('coms.accounts', [ ircAccount, xmppAccount ]);
+    this.set('selectedAccountId', ircAccount.id);
+    await render(hbs`<JoinChannel />`);
+
+    await select('select#account', xmppAccount.id);
+    assert.equal(this.selectedAccountId, xmppAccount.id, 'updates the selected account ID property');
```

Comment:
This works in the browser, but I can't get the test to succeed.

@galfert Maybe you have an idea?

Reply:
In integration tests you can't access the internal properties of the component. The `this` is the outer context of the component, e.g. the controller that uses the component.

Usually you would pass an action to a component and then test that it's called with the correct argument(s). Or you test that the DOM changes according to the interaction with the component.

I think in this case the appropriate test would be to also click the join button and then test that `coms.createChannel` is called with the correct arguments.

I'll have a stab at that.